packages/Search/test/Search_UnitTestHelpers.hpp

```diff
@@ -0,0 +1,220 @@
+                           std::vector<DataTransferKit::Box> const &b )
+{
+    int const n = b.size();
+    Kokkos::View<DataTransferKit::Box *, DeviceType> boxes( "boxes", n );
+    auto boxes_host = Kokkos::create_mirror_view( boxes );
+    for ( int i = 0; i < n; ++i )
+        boxes_host( i ) = b[i];
+    Kokkos::deep_copy( boxes, boxes_host );
+    return DataTransferKit::DistributedSearchTree<DeviceType>( comm, boxes );
+}
+
+// Hopefully we can git rid of this once the operator== has been implemented
+// for boxes.
+void testBoxEquality( DataTransferKit::Box const &l,
+                      DataTransferKit::Box const &r, bool &success,
+                      Teuchos::FancyOStream &out )
+{
+    TEST_EQUALITY( l[0], r[0] );
+    TEST_EQUALITY( l[1], r[1] );
+    TEST_EQUALITY( l[2], r[2] );
+    TEST_EQUALITY( l[3], r[3] );
+    TEST_EQUALITY( l[4], r[4] );
+    TEST_EQUALITY( l[5], r[5] );
+}
+
+template <typename DeviceType>
+Kokkos::View<DataTransferKit::Details::Overlap *, DeviceType>
+makeOverlapQueries( std::vector<DataTransferKit::Box> const &boxes )
+{
+    int const n = boxes.size();
+    Kokkos::View<DataTransferKit::Details::Overlap *, DeviceType> queries(
+        "overlap_queries", n );
+    auto queries_host = Kokkos::create_mirror_view( queries );
+    for ( int i = 0; i < n; ++i )
+        queries_host( i ) = DataTransferKit::Details::overlap( boxes[i] );
+    Kokkos::deep_copy( queries, queries_host );
+    return queries;
+}
+
+template <typename DeviceType>
+Kokkos::View<DataTransferKit::Details::Nearest *, DeviceType>
+makeNearestQueries(
+    std::vector<std::pair<DataTransferKit::Point, int>> const &points )
+{
+    // NOTE: `points` is not a very descriptive name here. It stores both the
+    // actual point and the number k of neighbors to query for.
+    int const n = points.size();
+    Kokkos::View<DataTransferKit::Details::Nearest *, DeviceType> queries(
+        "nearest_queries", n );
+    auto queries_host = Kokkos::create_mirror_view( queries );
+    for ( int i = 0; i < n; ++i )
+        queries_host( i ) = DataTransferKit::Details::nearest(
+            points[i].first, points[i].second );
+    Kokkos::deep_copy( queries, queries_host );
+    return queries;
+}
+
+template <typename DeviceType>
+Kokkos::View<DataTransferKit::Details::Within *, DeviceType> makeWithinQueries(
+    std::vector<std::pair<DataTransferKit::Point, double>> const &points )
+{
+    // NOTE: `points` is not a very descriptive name here. It stores both the
+    // actual point and the radius for the search around that point.
+    int const n = points.size();
+    Kokkos::View<DataTransferKit::Details::Within *, DeviceType> queries(
+        "within_queries", n );
+    auto queries_host = Kokkos::create_mirror_view( queries );
+    for ( int i = 0; i < n; ++i )
+        queries_host( i ) = DataTransferKit::Details::within(
+            points[i].first, points[i].second );
```

Comment:
What happens if  `points[i].second < 0`?

Reply:
Same here, the predicate will not complain and will always evaluate all nodes to `false`.  The query on the tree will only process the root node and will exit directly after since the predicate won't be satisfied on the children (see [here](https://github.com/ORNL-CEES/DataTransferKit/blob/master/packages/Search/src/details/DTK_DetailsPredicate.hpp#L86-L91) and [here](https://github.com/ORNL-CEES/DataTransferKit/blob/master/packages/Search/src/details/DTK_DetailsTreeTraversal.hpp#L105-L130)). We may check the predicate on the root before adding it to the stack if you like but I don't think it is a big deal.